### PR TITLE
go/store/nbs: make chunk journal lock NBS manifest while open

### PR DESCRIPTION
**Pull request by andy-wm-arthur**

### CODE REVIEW

File: go/store/nbs/file_manifest_test.go [hunk 1]
```diff
@@ -77,6 +79,7 @@ func TestFileManifestLoadIfExists(t *testing.T) {
 }
 
 func TestFileManifestLoadIfExistsHoldsLock(t *testing.T) {
+	t.Skip("TODO")
```
Comment (andy-wm-arthur) [inline go/store/nbs/file_manifest_test.go:82]: this test fails now that we don't hold the lock while reading

Comment (reltuk) [reply go/store/nbs/file_manifest_test.go:82]: Remove it? Its name seems to imply it's testing for behavior we never intend to implement.

File: go/store/nbs/journal.go [hunk 2]
```diff
@@ -407,6 +419,117 @@ func (c journalConjoiner) chooseConjoinees(upstream []tableSpec) (conjoinees, ke
...
+		lock, err = nil, nil // read only
+	} else if err != nil {
+		return nil, err
+	}
+	m = &journalManifest{dir: dir, lock: lock}
+
+	var f *os.File
+	f, err = openIfExists(filepath.Join(dir, manifestFileName))
+	if err != nil {
+		return nil, err
```
Comment (reltuk) [inline go/store/nbs/journal.go:438]: I think we want to unlock m.lock here and in other error paths. Thoughts?

File: integration-tests/bats/sql-client.bats [hunk 3]
```diff
@@ -37,7 +37,7 @@ setup() {
 }
 
 teardown() {
-    stop_sql_server
+    stop_sql_server 1 && sleep 0.5
```
Comment (reltuk) [inline integration-tests/bats/sql-client.bats:40]: Nit: Can we `wait` the pid we killed, or is this send metrics nonsense? A comment about why?

**Review by reltuk — APPROVED** [review]
LGTM! A few suggestions.

File: integration-tests/bats/sql-client.bats (hunk 3 above)
Comment (andy-wm-arthur) [reply integration-tests/bats/sql-client.bats:40]: I think this was `send-metrics` nonsense, and is now fixed

**Pull request merged**